File: src/Tests4/Circle.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Tests4
{
    class Circle : Curve
    {
        double Radius;

       public Circle(int x, int y, double radius) : base(x, y)
        {
            Radius = radius;
        }

        public double getRadius()
        {
            return Radius;
        }

        public override bool ClosedCurve()
        {
            return true;
        }

        public override Point PointCalcul(int t)
        {
            return new Point(Radius * Math.Cos(t),
                Radius * Math.Sin(t));
        }

        public override double TangentCalcul(int t)
        {
            Point pt = PointCalcul(t);
            double f = -(Math.Cos(t) / Math.Sin(t));
            // общий вид касательной
            // y-y0=f'(x-x0)
            Console.WriteLine("y "+(-pt.getY())+" = "+f+" (x"+-(-pt.getX())+" )");
            return 0;
        }
    }
}

File: src/Circle.cpp
// Include self.
#include <Circle.h>

// Includes from STL.
#include <math.h>

// Includes from Core.
#include <Curve.h>
#include <Vector.h>
#include <Point.h>

//=============================================================================
//function : Gradient
//purpose  : находит вектор касательной к кругу
//=============================================================================
Vector Circle::Gradient(double t) const
{
  double x = Radius * (-sin(t));
  double y = Radius * cos(t);
  return Vector(x, y);
}

//=============================================================================
//function : PointCalcul
//purpose  : возвращает точку в зависимости от параметра.
//=============================================================================
Point Circle::PointCalcul(double t) const
{
  return Point(point.getX() + Radius * cos(t),
               point.getY() + Radius * sin(t));
}
